src/components/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useCallback, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { FlexContainer } from '../FlexContainer';
+import { AccordionProps } from './Accordion.types';
+import AccordionCollapsible from './AccordionCollapsible';
+
+const Accordion: React.FC<AccordionProps> = ({
+  isCollapsedOnOpen = true,
+  items,
+  ...props
+}) => {
+  const [openIndexes, setOpenIndexes] = useState(
+    items.filter((item) => item.isOpen).map((item, index) => index),
+  );
+
+  const handleClick = useCallback(
+    (index: number) => {
+      setOpenIndexes(
+        openIndexes.includes(index)
+          ? openIndexes.filter((i) => i !== index)
+          : isCollapsedOnOpen
+          ? [index]
+          : [...openIndexes, index],
```

Comment:
Just my preference but nested ternaries are hell to read 😄

Reply:
Yeah, wondered if it's gonna catch someone's eyes. Can break it into sub call or something.

Reply:
I think it's used all over ui-core anyway so doesn't matter to me much